### PR TITLE
PBM-734: add storage.s3.debugLogLevels option

### PR DESCRIPTION
**Pull request by defbin**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Sharded cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 4.4 | passed |
| Sharded cluster mongodb 5.0 | failed |
| Non-sharded replicaset mongodb 4.4 | passed |
| Non-sharded replicaset mongodb 4.0 | passed |
| Single-node replicaset mongodb 4.4 | passed |
| Restore on new cluster mongodb 4.4 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Non-sharded replicaset mongodb 4.2 | passed |
| Single-node replicaset mongodb 4.0 | passed |
| Restore on new cluster mongodb 5.0 | passed |
| Single-node replicaset mongodb 5.0 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Single-node replicaset mongodb 4.2 | passed |
| Non-sharded replicaset mongodb 5.0 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: pbm/storage/s3/s3.go [hunk 1]
```diff
@@ -88,6 +125,42 @@ func (c *Conf) Cast() error {
...
+
+	for _, lvl := range strings.Split(levels, ",") {
+		l := SDKDebugLogLevel(lvl).SDKLogLevel()
+		if l == 0 {
+			fmt.Fprintf(out, "WARN: S3 client debug log level: unsupported %q\n", lvl)
+			continue
+		}
+
+		if _, ok := ss[lvl]; ok {
+			fmt.Fprintf(out, "WARN: S3 client debug log level: duplicated %q\n", lvl)
```
Comment (dAdAbird) [inline pbm/storage/s3/s3.go:153]: s/WARN/Warning
In other places in CLI we are using the`"Error:.."` format. `"Warning: ..."` would be more consistent

**Review by dAdAbird — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Sharded cluster mongodb 4.2 | failed |
| Sharded cluster mongodb 4.0 | failed |
| Sharded cluster mongodb 4.4 | failed |
| Sharded cluster mongodb 5.0 | failed |
| Non-sharded replicaset mongodb 4.0 | passed |
| Restore on new cluster mongodb 5.0 | passed |
| Non-sharded replicaset mongodb 4.4 | passed |
| Non-sharded replicaset mongodb 5.0 | passed |
| Single-node replicaset mongodb 4.4 | failed |
| Restore on new cluster mongodb 4.0 | passed |
| Non-sharded replicaset mongodb 4.2 | failed |
| Single-node replicaset mongodb 4.0 | failed |
| Single-node replicaset mongodb 5.0 | failed |
| Single-node replicaset mongodb 4.2 | failed |
| Restore on new cluster mongodb 4.2 | passed |
| Restore on new cluster mongodb 4.4 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Single-node replicaset mongodb 4.2 | passed |
| Single-node replicaset mongodb 4.4 | passed |
| Single-node replicaset mongodb 4.0 | passed |
| Non-sharded replicaset mongodb 4.2 | passed |
| Non-sharded replicaset mongodb 4.4 | failed |
| Non-sharded replicaset mongodb 5.0 | failed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Single-node replicaset mongodb 5.0 | failed |
| Restore on new cluster mongodb 5.0 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Restore on new cluster mongodb 4.4 | passed |
| Sharded cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 5.0 | failed |
| Sharded cluster mongodb 4.4 | passed |

File: pbm/storage/s3/s3.go [hunk 2]
```diff
@@ -88,6 +125,42 @@ func (c *Conf) Cast() error {
...
+	}
+
+	if out == nil {
+		out = ioutil.Discard
+	}
+
+	var logLevel aws.LogLevelType
+	ss := make(map[string]struct{})
+
+	for _, lvl := range strings.Split(levels, ",") {
```
[the remark this answers is not in the document]
Comment (defbin) [reply pbm/storage/s3/s3.go:145]: I understand it. My point was to inform users if they made a mistake. IMO any feedback from a tool is good UX and part of the interface. For me, warning let me know something could work not I expect or ask, e.g., modified after applying like with `" HTTPBody " => "HTTPBody"`. BTW the original passed value is stored in config without corrections.

Anyway, I'll remove the warnings and only leave about an unsupported level type.

File: pbm/config.go [hunk 3]
```diff
@@ -178,6 +179,8 @@ func (p *PBM) SetConfig(cfg Config) error {
 		if err != nil {
 			return errors.Wrap(err, "cast storage")
 		}
+
+		s3.SDKLogLevel(cfg.Storage.S3.DebugLogLevels, os.Stderr)
```
[the remark this answers is not in the document]
Comment (defbin) [reply pbm/config.go:183]: I didn't get why we need the comment?  `s3.SDKLogLevel()` has a doc comment that describes how it could use the `os.Stderr`.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Sharded cluster mongodb 4.0 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 4.4 | failed |
| Sharded cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 5.0 | failed |
| Non-sharded replicaset mongodb 4.2 | failed |
| Non-sharded replicaset mongodb 4.0 | passed |
| Restore on new cluster mongodb 4.4 | passed |
| Non-sharded replicaset mongodb 4.4 | passed |
| Restore on new cluster mongodb 5.0 | passed |
| Non-sharded replicaset mongodb 5.0 | failed |
| Single-node replicaset mongodb 4.0 | passed |
| Single-node replicaset mongodb 4.4 | passed |
| Single-node replicaset mongodb 4.2 | passed |
| Single-node replicaset mongodb 5.0 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Sharded cluster mongodb 4.2 | failed |
| Sharded cluster mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.4 | failed |
| Restore on new cluster mongodb 4.4 | passed |
| Non-sharded replicaset mongodb 4.2 | failed |
| Restore on new cluster mongodb 5.0 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Restore on new cluster mongodb 4.0 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Sharded cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 5.0 | passed |
| Sharded cluster mongodb 4.4 | passed |
| Sharded cluster mongodb 4.2 | passed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 5.0 | passed |
| Non-sharded replicaset mongodb 4.4 | passed |
| Single-node replicaset mongodb 5.0 | passed |
| Single-node replicaset mongodb 4.2 | passed |
| Restore on new cluster mongodb 5.0 | passed |
| Non-sharded replicaset mongodb 4.2 | passed |
| Restore on new cluster mongodb 4.4 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Single-node replicaset mongodb 4.4 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Single-node replicaset mongodb 4.0 | passed |

**Pull request merged**